src/sync/deployment.test.ts

```diff
@@ -30,6 +37,221 @@ describe("sync/deployments", () => {
+
+			repoSyncState = dbState.repoSyncState!;
+			repositoryData = {
+				id: repoSyncState.repoId,
+				name: repoSyncState.repoName,
+				full_name: repoSyncState.repoFullName,
+				owner: { login: repoSyncState.repoOwner },
+				html_url: repoSyncState.repoUrl,
+				updated_at: repoSyncState.repoUpdatedAt?.toISOString()
+			};
+
+			"*".repeat(REPEAT_LOTS_OF_TIME).split("").forEach(() => githubUserTokenNock(installationId));
+		});
+
+		afterEach(async () => {
+			nockCleanAll();
+		});
+
+		describe("when ff is on", () => {
+
+			beforeEach(() => {
+				when(booleanFlag).calledWith(BooleanFlags.USE_DYNAMODB_FOR_DEPLOYMENT_BACKFILL, jiraHost).mockResolvedValue(true);
+			});
+
+			// eslint-disable-next-line jest/expect-expect
+			it("should save deployments to dynamodb and process WITHOUT calling rest listing api", async () => {
+
+				const deployments = createDeploymentEntities(4);
+
+				nockFetchingDeploymentgPagesGraphQL(DEPLOYMENT_CURSOR_EMPTY, [deployments[3], deployments[2]]);
+				nockFetchingDeploymentgPagesGraphQL(deployments[2].cursor, [deployments[1], deployments[0]]); //this is for extra page when fetching current deployments
+
+				nockDeploymentCommitGetApi([deployments[3], deployments[2]], REPEAT_ONCE);
+
+				const result = await getDeploymentTask(logger, gitHubClient, jiraHost, repositoryData, DEPLOYMENT_CURSOR_EMPTY, PAGE_SIZE__TWO_ITEMS, msgPayload());
+
+				await expectDeploymentEntryInDB([deployments[3], deployments[2]]);
+				expectEdgesAndPayloadMatchToDeploymentCommits(result, [deployments[3], deployments[2]]);
+
+			});
+		});
+
+		describe("when ff is off", () => {
+
+			beforeEach(() => {
+				when(booleanFlag).calledWith(BooleanFlags.USE_DYNAMODB_FOR_DEPLOYMENT_BACKFILL, jiraHost).mockResolvedValue(false);
+			});
+
+			describe("for empty demployment cursor", () => {
+				// eslint-disable-next-line jest/expect-expect
+				it("should fetch deployments from begining", async () => {
+
+					const deployments = createDeploymentEntities(4);
+
+					nockFetchingDeploymentgPagesGraphQL(DEPLOYMENT_CURSOR_EMPTY, [deployments[3], deployments[2]]);
+
+					nockDeploymentCommitGetApi([deployments[3], deployments[2]], REPEAT_ONCE);
+
+					//because ff is off, it won't find cache in history, so need to call the rest listing api lots of times
+					nockDeploymentListingApi(deployments, REPEAT_LOTS_OF_TIME);
+					nockDeploymentStatusApi(deployments, REPEAT_LOTS_OF_TIME);
+
+					const result = await getDeploymentTask(logger, gitHubClient, jiraHost, repositoryData, DEPLOYMENT_CURSOR_EMPTY, PAGE_SIZE__TWO_ITEMS, msgPayload());
+
+					expectEdgesAndPayloadMatchToDeploymentCommits(result, [deployments[3], deployments[2]]);
+				});
+			});
+
+			describe("for existing deployment cursor", () => {
+				// eslint-disable-next-line jest/expect-expect
+				it("should fetch deployments from existing cursor", async () => {
+
+					const deployments = createDeploymentEntities(4);
```

Comment:
Can we convert the 'magic number' of 4 to a variable so it's more explicit and gives more context? e.g.

```
 const deploymentCount = 4;
  const deployments = createDeploymentEntities(deploymentCount);
```

Reply:
fixed.